### ISSUE TITLE
[hw,i2c] Unexpected assertion of intr_sda_unstable_o interrupt

### ISSUE THREAD

**Issue by crteja** [lowRISC/opentitan#17811]
### Description

With https://github.com/lowRISC/opentitan/pull/17809, a new check is added for `intr_sda_unstable_o`, `intr_scl_interference_o` and `intr_sda_interference_o` interrupts. 
With this change, there are a number of `i2c_host_error_intr` has many failures with unexpected `intr_sda_unstable_o` interrupt accounting for most number of failures.
From the waveform, there doesn't seem to be any interference on SDA signal.
![image](https://user-images.githubusercontent.com/124047397/229847314-7cd964b8-e33c-45b1-8901-26b21f9225d9.png)

Upon further debug, in `i2c_fsm.sv`, sda_interference is driven by an assignment statement
![image](https://user-images.githubusercontent.com/124047397/229847607-fd7f97f1-4224-4397-aa29-f135f572f0d5.png)
the condition is satisfied for the first expression, `host_enable_i` is set to 1 since host mode is enabled. `host_idle_o` is asserted in the next cycle after stop condition is detected. But `sda_i` is delayed by two cycles at the boundary of `i2c_fsm.sv`. Since the condition is satisfied for one cycle, design is asserting `intr_sda_interference_o`

![image](https://user-images.githubusercontent.com/124047397/229850030-01a65cdc-b801-4d42-89f8-580434263239.png)

Issue can be reproduced with https://github.com/lowRISC/opentitan/pull/17809, 
`util/dvsim/dvsim.py hw/ip/i2c/dv/i2c_sim_cfg.hjson  -i   i2c_host_error_intr --tool xcelium  --fixed-seed 3743598109 --waves shm`

**Comment by marnovandermaas** [comment]
Based on the following comment, this event should only be triggered when the host is first enabled:
https://github.com/lowRISC/opentitan/blob/f1e4f2dae1ca51f8d4d84bfe3aea063da6da54d1/hw/ip/i2c/rtl/i2c_fsm.sv#L423

We could introduce a flop to detect a rising edge in `host_enable_i` and only drive the event if that is true.

**Comment by msfschaffner** [comment]
Discussed with @moidx and @jesultra and it has been determined that this bug is low risk and can be deferred to after M2.5.

There is a software workaround for this (disable the spurious interrupts).

**Comment by andreaskurth** [comment]
@a-will: This has been raised quite some time ago. Given your current work, do you know if this is still an issue?

**Comment by a-will** [comment]
In https://github.com/lowRISC/opentitan/pull/22872, I remove that first condition. It seems to me like overreach and is inconsistent with the definition. I've pulled SDA interference back to actual loss of arbitration types.

If the user wishes to verify that SDA isn't low before starting a transaction, that user can simply read the `VAL` CSR for the current oversampled values.

In single-controller environments, SDA interference should trigger immediately upon attempting a transaction anyway. We should verify that this behavior does occur in DV, though.

**Comment by andreaskurth** [comment]
Thx @a-will. So essentially this needs to be covered by DV. Even if this was broken, would there still be a SW workaround (as mentioned earlier)? Then we could deprioritize this to M5.

**Comment by a-will** [comment]
At this point, the premise of the changes are wrong: "SDA interference" and "SDA unstable" will no longer be completely independent after the multi-controller support merges.

I have redefined these in the multi-controller PR:
- "SDA unstable" triggers if the value of SDA changes while SCL is high. This means we received an unexpected Start or Stop condition.
- "SDA interference" triggers if the i2c IP's controller is "transmitting" high, but when SCL is high, the bus shows SDA is low.

So, if the controller module is "transmitting" a logic high, an observed falling edge on SDA while SCL is high would trigger both!

I think we can deprioritize testing that the "SDA unstable" interrupt only triggers on specific conditions and no others, though--The interrupt can simply be disabled. It has questionable value.

On the other hand, it's critical to ensure any form of arbitration loss is correctly handled by the FSM (by setting CONTROLLER_EVENTS.ARBITRATION_LOST and halting the FSM). We don't have to get the _attribution_ to components absolutely correct, but "SDA interference" and "SDA unstable" are both triggering conditions.

**Comment by a-will** [comment]
Okay, I will close this issue for these reasons:
- The original failing case will no longer apply after the multi-controller PR closes.
- The code that drives the original failing case is invalid, as it assumes "SDA interference" and "SDA unstable" are independent, but they aren't. Both interrupts should trigger if a target pulls SDA low while SCL is high and the internal controller is attempting to transmit high. The agent can cause this to happen with tSdaInterference being nonzero.

Please re-open if you disagree.
